### PR TITLE
World generation events

### PR DESCRIPTION
**Pull request by BoogieMonster1O1**
Adds worldgen related events

- [x] Grass color event
- [x] Water color event
- [x] Foliage color event
- [x] Ore registration event
- [x] Ore generation cancelling event
- [x] Biome decoration event

### CODE REVIEW

**Comment by BoogieMonster1O1** [comment]
gonna yeet this and redo it next year

**Comment by BoogieMonster1O1** [comment]
yes it bacc

**Review by BluCobalt — CHANGES_REQUESTED** [review]
make build not fail

File: fabric-api-base/src/main/java/net/fabricmc/fabric/api/util/IntActionResult.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
```
Comment (BluCobalt) [inline fabric-api-base/src/main/java/net/fabricmc/fabric/api/util/IntActionResult.java]: shouldn't the copyright say legacy fabric?

File: fabric-world-generation-events-v1/src/main/java/net/fabricmc/fabric/api/worldgen/event/v1/BiomeDecorationEvents.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
```
Comment (BluCobalt) [inline fabric-world-generation-events-v1/src/main/java/net/fabricmc/fabric/api/worldgen/event/v1/BiomeDecorationEvents.java]: legacy fabric?

File: fabric-world-generation-events-v1/src/main/java/net/fabricmc/fabric/api/worldgen/event/v1/OreEntry.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
```
Comment (BluCobalt) [inline fabric-world-generation-events-v1/src/main/java/net/fabricmc/fabric/api/worldgen/event/v1/OreEntry.java]: lf?

File: fabric-world-generation-events-v1/src/main/java/net/fabricmc/fabric/api/worldgen/event/v1/OreRegistry.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
```
Comment (BluCobalt) [inline fabric-world-generation-events-v1/src/main/java/net/fabricmc/fabric/api/worldgen/event/v1/OreRegistry.java]: bruh

File: fabric-world-generation-events-v1/src/main/java/net/fabricmc/fabric/api/worldgen/event/v1/client/BiomeColorEvents.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
```
Comment (BluCobalt) [inline fabric-world-generation-events-v1/src/main/java/net/fabricmc/fabric/api/worldgen/event/v1/client/BiomeColorEvents.java]: unless this is being copy + pasted from upstream this should say legacy fabric

File: fabric-world-generation-events-v1/src/main/java/net/fabricmc/fabric/impl/worldgen/event/OreCancellable.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
```
Comment (BluCobalt) [inline fabric-world-generation-events-v1/src/main/java/net/fabricmc/fabric/impl/worldgen/event/OreCancellable.java]: i cant do this for each one

**Review by BluCobalt — CHANGES_REQUESTED** [review]
ubrh

**Review by BluCobalt — APPROVED** [review]
haha oops

**Comment by BoogieMonster1O1** [comment]
forgot a testmod, will make one soon

**Comment by BoogieMonster1O1** [comment]
this is broken anyway lel

**Comment by Dolphin2Point1** [comment]
why tf are you doing this again this is dumb

**Comment by BoogieMonster1O1** [comment]
> why tf are you doing this again this is dumb

ikr its so dumb

**Pull request closed without merging**